utils.go

```diff
@@ -413,16 +414,28 @@ func WriteCgroupProc(dir string, pid int) error {
 	return err
 }
 
-// Since the OCI spec is designed for cgroup v1, in some cases
-// there is need to convert from the cgroup v1 configuration to cgroup v2
-// the formula for cpuShares is y = (1 + ((x - 2) * 9999) / 262142)
-// convert from [2-262144] to [1-10000]
-// 262144 comes from Linux kernel definition "#define MAX_SHARES (1UL << 18)"
+// ConvertCPUSharesToCgroupV2Value converts CPU shares, used by cgroup v1,
+// to CPU weight, used by cgroup v2.
 func ConvertCPUSharesToCgroupV2Value(cpuShares uint64) uint64 {
+	// The value of 0 means "unset".
 	if cpuShares == 0 {
 		return 0
 	}
-	return (1 + ((cpuShares-2)*9999)/262142)
+	// Cgroup v1 CPU shares has a range of [2^1...2^18], i.e. [2...262144],
+	// and the default value is 1024.
+	// Cgroup v2 CPU weight has a range of [10^0...10^4], i.e. [1...10000],
+	// and the default value is 100.
+	if cpuShares < 2 {
+		return 1
+	}
+	if cpuShares > 262144 {
```

Comment:
Really nitty, but these could be `<=` and `>=` to avoid doing the calculation

Reply:
Done.

Although this also results in formula not being used for boundary values, so we're no longer testing it giving min and max as inputs. In order to test that we now need to add sharesMin+1 and sharesMax-1 values to the test.

Added as well.